### DISCUSSION TITLE
Prefix by Account

### DISCUSSION THREAD

**Discussion by jeff282** [burkybang/SuiteAdvanced-Feedback#89] [Ideas]
I would love if we could make ID prefixes by account, similar to how colours are setup. I regularly work across client accounts and want to use a different prefix specific to that client in each. 

Even better would be a global default prefix as well as a list of account specific. If no account specific exists then use global etc.

**Comment by burkybang** [comment]
Thanks for the feedback. May I ask why you prefix script IDs with the client? What is the purpose of doing that?

**Reply by jeff282** [reply]
This is so any custom elements can be easily identified and are specific to their instance. I work at an implementation partner and clients don't necessarily want a prefix related to us used, they may align on a prefix for all of their custom elements that is unique to them. Typically we use the initials of the company name.

**Reply by burkybang** [reply]
Why use a prefix at all if the custom element is specific to the current account?

**Reply by jeff282** [reply]
This is so it's easy to identify the elements we've created and need to potentially migrate or document etc. There is also likely other custom elements created by NS provided bundles or other solution providers and suiteapps, so it helps differentiate.

**Reply by burkybang** [reply]
What if other consultants use the same prefix for the current account? If everyone prefixes custom objects with the client, then it wouldn't make it easier to tell which ones have been created by your company.

These are the only situations where I think prefixes are valuable:
1. Objects created by a consultant using the consultant's prefix (Unless the client prefers otherwise of course)
2. Objects packaged in a SuiteApp or SuiteBundle that will be installed in many different accounts

In my opinion, I can't see the value of client prefixes, but I'm happy to discuss further.

**Reply by jeff282** [reply]
Fair concern re other consultants, however we typically are the only ones in the NetSuite instance (fairly large consulting team).

I see the validity of your points, that said in my experience both working at NetSuite itself and externally using client prefixes has been standard practice.

**Reply by burkybang** [reply]
Yeah, I've seen it used often. I think it's "standard" because it's popular, but that doesn't mean it's actually useful 🤷‍♂️

**Comment by NicoleAnswer** [comment]
I actually use both options.  I have one client that specifically wants the work i do to be their prefix, but thats because my relationship with them is different to other clients.  In an ideal work you could set the prefix for each client, but have a checkbox on the settings to allow you to use the primary Unless its specifically over ridden.  This would give the best of both worlds.

**Reply by burkybang** [reply]
Thanks for your input!

My main hesitation for implementing something like this is Chrome's sync storage limit. It's already possible to hit the capacity by setting a custom theme for too many NetSuite accounts. (I know. I've done it.) Offering the ability to set a prefix for each one will surely use up the storage quicker.

I could instead use local storage, which has a much higher limit, but then the settings wouldn’t sync across computers.

If I didn't have this limitation to deal with, I'd have so many more settings!

**Reply by eduardRsult** [reply]
Most of the time If the client is the owner of the customisation then the client requires a specific prefix. Especially if there are multiple consultant companies working on the same account. Then it is not up to us. 
In other scenarios there is a double prefix. 
Then again hvaing prefixes per account would make that more seamless. 
Its not a big thing since you can edit the id after generation. If there would not be a storage limit this would definitely be useful.
Thanks for telling the reason behind your hesitation.
I am sure you are looking into ways to circumvent that limitation.
Splitting setttings up in synced settings and non-synced settings might be confusing. Maybe there is a natural split somewhere.
After all using different computers is for some of us only done once in a longer while.
